src/MapViewController.swift

```diff
@@ -417,11 +476,184 @@ open class MapViewController: UIViewController, LocationManagerDelegate, TGRecog
+    self.loadDelegate?.mapController!(self, didLoadSceneAsync: scene)
+  }
+  
+  open func mapViewDidCompleteLoading(_ mapView: TGMapViewController) {
+    guard (self.loadDelegate != nil) && (self.loadDelegate?.responds(to: #selector(MapLoadDelegate.mapControllerDidCompleteLoading(_:))))! else {
+      return
+    }
+    self.loadDelegate?.mapControllerDidCompleteLoading!(self)
+  }
+  
+  open func mapView(_ mapView: TGMapViewController, didSelectFeature feature: [AnyHashable : Any]?, atScreenPosition position: CGPoint) {
+    guard (feature != nil) else {
+      return
+    }
+    guard (self.selectDelegate != nil) else {
+      return
+    }
+    guard (self.selectDelegate?.responds(to: #selector(MapSelectDelegate.mapController(_:didSelectFeature:atScreenPosition:))))! else {
+      return
+    }
+    self.selectDelegate?.mapController!(self, didSelectFeature: feature, atScreenPosition: position)
+  }
+  
+  open func mapView(_ mapView: TGMapViewController, didSelectLabel labelPickResult: TGLabelPickResult?, atScreenPosition position: CGPoint) {
+    guard (labelPickResult != nil) else {
+      return
+    }
+    guard (self.selectDelegate != nil) else {
+      return
+    }
+    guard (self.selectDelegate?.responds(to: #selector(MapSelectDelegate.mapController(_:didSelectLabel:atScreenPosition:))))! else {
+      return
+    }
+    self.selectDelegate?.mapController!(self, didSelectLabel: labelPickResult, atScreenPosition: position)
+  }
+  
+  open func mapView(_ mapView: TGMapViewController, didSelectMarker markerPickResult: TGMarkerPickResult?, atScreenPosition position: CGPoint) {
+    guard (markerPickResult != nil) else {
+      return
+    }
+    guard (self.selectDelegate != nil) else {
+      return
+    }
+    guard (self.selectDelegate?.responds(to: #selector(MapSelectDelegate.mapController(_:didSelectMarker:atScreenPosition:))))! else {
+      return
+    }
+    self.selectDelegate?.mapController!(self, didSelectMarker: markerPickResult, atScreenPosition: position)
+  }
+  
+  //MARK : TGRecognizerDelegate
+  
+  open func mapView(_ view: TGMapViewController, recognizer: UIGestureRecognizer, shouldRecognizeSingleTapGesture location: CGPoint) -> Bool {
+    guard (self.gestureDelegate != nil) else {
+      return true
+    }
+    guard (self.gestureDelegate?.responds(to: #selector(MapGestureDelegate.mapController(_:recognizer:shouldRecognizeSingleTapGesture:))))! else {
+      return true
+    }
+    let recognize = (self.gestureDelegate?.mapController!(self, recognizer: recognizer, shouldRecognizeSingleTapGesture: location))!
+    if !recognize {
+      self.tgViewController.pickLabel(at: location)
+      self.tgViewController.pickMarker(at: location)
+      self.tgViewController.pickFeature(at: location)
+    }
+    return recognize
+  }
+  
+  open func mapView(_ view: TGMapViewController, recognizer: UIGestureRecognizer, didRecognizeSingleTapGesture location: CGPoint) {
+    self.tgViewController.pickLabel(at: location)
```

Comment:
Maybe I'm misunderstanding the question, but I think it is the right behavior. It is up to the user of the framework to decide whether a callback on a touch event triggers a selection of makers/labels etc.. Nothing relating to selection callback is automatically called on the Tangram side except if a `pick*` call happens. This seems to be quite similar to what happens in the [android SDK](https://github.com/mapzen/android/blob/6e52222a8b39c77e5b6cd795b67090668f94c4db/core/src/main/java/com/mapzen/android/graphics/MapzenMap.java#L59).